### PR TITLE
Dlockhart/lit2 preparations

### PR DESCRIPTION
**Pull request by dlockhart**
Just cleaning some stuff up before the Lit 2 migration:

- Switched to use the newer `@web/test-runner`
- Switched to use the newer `@web/dev-server`
- Removed the old web component polyfill (not needed anymore after legacy-Edge EOL)

I also did a fair bit of cleanup in the Lit components. Many of them were extending the HMC foundation stuff but not using any of its features. This was causing a lot of `d2l-fetch` calls to nowhere for no reason. Similarly, some were using the localize mixin but didn't have any localization hooked up.

### CODE REVIEW

File: src/d2l-flashcards-color-picker.js [hunk 1]
```diff
@@ -1,43 +1,9 @@
...
-				type: String,
-				observable: observableTypes.property
-			},
-			cardId: {
-				type: String,
-				observable: observableTypes.property
-			},
-			opened: {
-				type: Boolean
-			}
```
Comment (dlockhart) [inline src/d2l-flashcards-color-picker.js:38]: It looks like these properties were just copy-pasted into most of the classes? Each class only needs to list its own properties.

File: src/d2l-flashcards-color-picker.js [hunk 2]
```diff
@@ -105,18 +71,6 @@ class ColorPicker extends HypermediaStateMixin(LocalizeDynamicMixin(LitElement))
...
-		super();
-		this.updatedDisabled = false;
-		this.addEventListener('change', this._onChange);
-	}
-
-	static get localizeConfig() {
-		return {
-			importFunc: async lang => (await import(`./lang/${lang}.js`)).default
-		};
-	}
```
Comment (dlockhart) [inline src/d2l-flashcards-color-picker.js:118]: If a component doesn't have any localization (i.e. a call to `this.localize()`), then it doesn't need this config and doesn't need to extend the mixin.

File: src/d2l-flashcards-color-picker.js [hunk 3]
```diff
@@ -132,7 +86,7 @@ class ColorPicker extends HypermediaStateMixin(LocalizeDynamicMixin(LitElement))
                             <button class="d2l-color-block d2l-yellow"></button>
                     </div>
                 </d2l-dropdown-content>
-            </d2l-dropdown-button>
+            </d2l-dropdown-button-subtle>
```
Comment (dlockhart) [inline src/d2l-flashcards-color-picker.js:89]: Wrong closing tag.

File: src/d2l-flashcards-deck-collection.js [hunk 4]
```diff
@@ -81,18 +50,6 @@ class FlashcardDeckCollection extends HypermediaStateMixin(LocalizeDynamicMixin(
 		`;
 	}
 
-	constructor() {
-		super();
-		this.updatedDisabled = false;
-		this.addEventListener('change', this._onChange);
```
Comment (dlockhart) [inline src/d2l-flashcards-deck-collection.js:87]: This code didn't seem to be doing anything -- `updateDisabled` isn't a property and `this._onChange` isn't defined.

File: src/d2l-flashcards-edit-dialog.js [hunk 5]
```diff
@@ -1,36 +1,16 @@
 import '@brightspace-ui/core/components/button/button.js';
 import '@brightspace-ui/core/components/dialog/dialog-fullscreen.js';
-import '@brightspace-ui/core/components/dialog/dialog.js';
+import '@brightspace-ui/core/components/inputs/input-textarea.js';
```
Comment (dlockhart) [inline src/d2l-flashcards-edit-dialog.js:3]: Be careful that you're importing the things you use and not importing other things. :)

File: src/d2l-flashcards-edit-dialog.js [hunk 6]
```diff
@@ -1,36 +1,16 @@
...
 import './d2l-flashcards-flashcard-flip.js';
 
 import { css, LitElement } from 'lit-element/lit-element.js';
-import { HypermediaStateMixin, observableTypes } from '@brightspace-hmc/foundation-engine/framework/lit/HypermediaStateMixin';
-import { html } from '@brightspace-hmc/foundation-engine/framework/lit/hypermedia-components';
-import { LocalizeDynamicMixin } from '@brightspace-ui/core/mixins/localize-dynamic-mixin';
+import { HypermediaStateMixin, observableTypes } from '@brightspace-hmc/foundation-engine/framework/lit/HypermediaStateMixin.js';
+import { html } from '@brightspace-hmc/foundation-engine/framework/lit/hypermedia-components.js';
+import { LocalizeDynamicMixin } from '@brightspace-ui/core/mixins/localize-dynamic-mixin.js';
```
Comment (dlockhart) [inline src/d2l-flashcards-edit-dialog.js:9]: Starting in Lit 2, all imports need `.js` extensions.

File: src/d2l-flashcards-edit-dialog.js [hunk 7]
```diff
@@ -140,20 +118,18 @@ class FlashcardEditDialog extends HypermediaStateMixin(LocalizeDynamicMixin(LitE
 			<d2l-dialog-fullscreen title-text="New Flashcard" ?opened="${this.opened}" @d2l-dialog-close="${this._handleClose}">
 				<div class="d2l-edit-row">
 					<div class="d2l-leftbox"> 
-						<div class="d2l-card-header">Front of Card</div>
-							<div class="d2l-text-container">
-								<d2l-input-textarea cardId="${this.cardId}" side="front" value="${this.localize('frontInputPlaceholder')}" placeholder="${this.localize('frontInputPlaceholder')}" @change=${this._onBackChange}></d2l-input-textarea>
```
Comment (dlockhart) [inline src/d2l-flashcards-edit-dialog.js:145]: These textareas were missing labels, so re-organized it a bit to use their built-in labels.

File: src/d2l-flashcards-edit-dialog.js [hunk 8]
```diff
@@ -140,20 +118,18 @@ class FlashcardEditDialog extends HypermediaStateMixin(LocalizeDynamicMixin(LitE
...
-							<div class="d2l-text-container">
-								<d2l-input-textarea cardId="${this.cardId}" side="back" value="${this.localize('backInputPlaceholder')}" placeholder="${this.localize('backInputPlaceholder')}" @change=${this._onBackChange}></d2l-input-textarea>
-							</div>
+						<div class="d2l-text-container">
+							<d2l-input-textarea label="Front of Card" cardId="${this.cardId}" side="front" value="${this.localize('frontInputPlaceholder')}" placeholder="${this.localize('frontInputPlaceholder')}" @change=${this._onBackChange}></d2l-input-textarea>
 						</div>
+						<div class="d2l-text-container">
+							<d2l-input-textarea label="Back of Card" cardId="${this.cardId}" side="back" value="${this.localize('backInputPlaceholder')}" placeholder="${this.localize('backInputPlaceholder')}" @change=${this._onBackChange}></d2l-input-textarea>
+						</div>
+					</div>
```
Comment (dlockhart) [inline src/d2l-flashcards-edit-dialog.js:127]: Missing `</div>` here.

File: src/d2l-flashcards-flashcard-flip.js [hunk 9]
```diff
@@ -132,8 +99,7 @@ class FlashcardFlip extends HypermediaStateMixin(LocalizeDynamicMixin(LitElement
 	`;
 	}
 	handleClick() {
-		const card = this.shadowRoot.querySelector('.d2l-card-inner');
-		card.classList.toggle('d2l-is-flipped');
+		this.flipped = !this.flipped;
```
Comment (dlockhart) [inline src/d2l-flashcards-flashcard-flip.js:102]: This is a perfect place to use a Lit property that reflects to a CSS attribute selector.

**Review by emsandrews — APPROVED** [review]

**Pull request merged**